go/teams/hidden/hidden.go

```diff
@@ -220,10 +220,73 @@ func generateKeyRotationSig3(mctx libkb.MetaContext, p GenerateKeyRotationParams
 	return &bun, ratchets, nil
 }
 
-func CheckFeatureGateForSupport(mctx libkb.MetaContext, teamID keybase1.TeamID, isWrite bool) (err error) {
+func CheckFeatureGateForSupportWithRotationType(mctx libkb.MetaContext, teamID keybase1.TeamID, isWrite bool, rt keybase1.RotationType) (ret keybase1.RotationType, err error) {
+	if rt == keybase1.RotationType_VISIBLE {
+		return rt, nil
+	}
+	ok, err := checkFeatureGateForSupport(mctx, teamID, isWrite)
+	if err != nil {
+		return rt, err
+	}
+
+	switch {
+	case rt == keybase1.RotationType_CLKR && !ok:
+		return keybase1.RotationType_VISIBLE, nil
```

Comment:
is this intended to return nil error?

Reply:
Yes. On prod, we're only allowing CLKRs to go through the hidden chain if it's explicitly allowed by the server.